ledger/sandbox/src/main/scala/com/digitalasset/platform/index/LedgerBackedIndexService.scala

```diff
@@ -104,38 +105,39 @@ abstract class LedgerBackedIndexService(
 
   // Returns a function that memoizes the current end
   // Can be used directly or shared throughout a request processing
-  private def convertOffset: LedgerOffset => Source[Offset, NotUsed] = {
-    lazy val currentEnd: Offset = ledger.ledgerEnd
-    domainOffset: LedgerOffset =>
-      domainOffset match {
-        case LedgerOffset.LedgerBegin => Source.single(Offset.beforeBegin)
-        case LedgerOffset.LedgerEnd => Source.single(currentEnd)
-        case LedgerOffset.Absolute(offset) =>
-          ApiOffset.fromString(offset).fold(Source.failed, off => Source.single(off))
-      }
+  private def convertOffset: LedgerOffset => Throwable \/ Offset = {
+    case LedgerOffset.LedgerBegin => \/-(Offset.beforeBegin)
+    case LedgerOffset.LedgerEnd => \/.fromTryCatchNonFatal(ledger.ledgerEnd)
+    case LedgerOffset.Absolute(offset) => \/.fromEither(ApiOffset.fromString(offset).toEither)
+  }
+
+  private def convertOffsetRange(
+      startExclusive: domain.LedgerOffset,
+      endInclusive: Option[domain.LedgerOffset]): Throwable \/ (Offset, Option[Offset]) = {
+    import scalaz.syntax.traverse._
+    import scalaz.std.option._
```

Comment:
@S11001001 
how do I get `map2` in scalaz?

Reply:
`^`, but for just tupling the results use `firstConversion tuple secondConversion`, which comes from `scalaz.syntax.apply._`.

Reply:
thanks!